backend/pages/api/ipfs/add-report.ts

```diff
@@ -0,0 +1,81 @@
+import type { NextApiRequest, NextApiResponse } from 'next';
+import Cors from 'cors';
+import { utils } from 'ethers';
+import { pinata } from '../../../services/ipfs';
+import { middleware } from '../../../middleware/middleware';
+import { postReport, postToSupportChannel } from '../../../services/discord';
+
+const MAX_MESSAGE_SIZE = parseInt(process.env.MAX_MESSAGE_SIZE || '4096');
+
+const cors = Cors({
+  methods: ['GET', 'HEAD'],
+});
+
+type Data = {
+  ipfsCid?: string;
+  ipfsUrl?: string;
+  message?: string;
+  detail?: string;
+};
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<Data>
+) {
+  await middleware(req, res, cors);
+
+  if (req.method !== 'POST') {
+    return res.status(405).json({ message: 'Method not allowed' });
+  }
+
+  const { address: addressString, sig, metadata: raw } = req?.body || {};
+
+  if (!raw) {
+    return res.status(422).json({ message: 'No message' });
+  }
+  // validate maximum size to prevent abuse
+  const metadataJSONString = JSON.stringify(raw);
+
+  if (metadataJSONString.length > MAX_MESSAGE_SIZE) {
+    return res.status(422).json({ message: 'Message too long' });
+  }
+
+  if (metadataJSONString.length === 0) {
+    return res.status(422).json({ message: 'No message' });
+  }
+
+  // validate address
+  if (!addressString || !utils.isAddress(addressString)) {
+    return res.status(422).json({ message: 'Missing or invalid address' });
+  }
+  const address = utils.getAddress(addressString);
+
+  postReport(raw.jobUrl, raw.reason);
```

Comment:
This will work for the MVP.

But it is prone to nuisance attacks if someone starts spamming the API.

Ideally we should handle this while syncing ReportJob log events from the API.